internal/engine/engine_test.go

```diff
@@ -45,62 +45,52 @@ func TestCheck(t *testing.T) {
 	eng, cancelFunc := mkEngine(t, param{auditLog: mockAuditLog, schemaEnforcement: schema.EnforcementNone})
 	defer cancelFunc()
 
-	ruleTableEngine, ruleTableCancelFunc := mkEngine(t, param{auditLog: mockAuditLog, schemaEnforcement: schema.EnforcementNone, enableRuleTable: true})
-	defer ruleTableCancelFunc()
-
-	engMap := map[string]*Engine{
-		"WithoutRuleEngine": eng,
-		"WithRuleEngine":    ruleTableEngine,
-	}
-
 	testCases := test.LoadTestCases(t, "engine")
 
-	for engPrefix, eng := range engMap {
-		for _, tcase := range testCases {
-			t.Run(fmt.Sprintf("%s/%s", engPrefix, tcase.Name), func(t *testing.T) {
-				tc := readTestCase(t, tcase.Input)
-				mockAuditLog.clear()
-
-				traceCollector := tracer.NewCollector()
-				haveOutputs, err := eng.Check(context.Background(), tc.Inputs, WithTraceSink(traceCollector))
+	for _, tcase := range testCases {
+		t.Run(tcase.Name, func(t *testing.T) {
+			tc := readTestCase(t, tcase.Input)
+			mockAuditLog.clear()
 
-				if tc.WantError {
-					require.Error(t, err)
-				} else {
-					require.NoError(t, err)
-				}
+			traceCollector := tracer.NewCollector()
+			haveOutputs, err := eng.Check(context.Background(), tc.Inputs, WithTraceSink(traceCollector))
 
-				for i, have := range haveOutputs {
-					// TODO(saml) I can't, for the life of me, figure out out to order this via a transformation
-					// function in `cmp.Diff` below, so this'll have to do for now
-					slices.SortStableFunc(have.Outputs, func(a, b *enginev1.OutputEntry) int {
-						if a.Src < b.Src {
-							return -1
-						} else if a.Src > b.Src {
-							return 1
-						}
-						return 0
-					})
-
-					require.Empty(t, cmp.Diff(tc.WantOutputs[i],
-						have,
-						protocmp.Transform(),
-						protocmp.SortRepeatedFields(&enginev1.CheckOutput{}, "effective_derived_roles"),
-					))
-				}
+			if tc.WantError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
 
-				haveDecisionLogs := mockAuditLog.getDecisionLogs()
+			for i, have := range haveOutputs {
+				// TODO(saml) I can't, for the life of me, figure out out to order this via a transformation
+				// function in `cmp.Diff` below, so this'll have to do for now
+				slices.SortStableFunc(have.Outputs, func(a, b *enginev1.OutputEntry) int {
+					if a.Src < b.Src {
+						return -1
+					} else if a.Src > b.Src {
+						return 1
+					}
+					return 0
+				})
 
-				require.Empty(t, cmp.Diff(tc.WantDecisionLogs,
-					haveDecisionLogs,
+				require.Empty(t, cmp.Diff(tc.WantOutputs[i],
+					have,
```

Comment:
Imports stdlib cmp: `import stdcmp "cmp"`

```suggestion
				require.Empty(t, cmp.Diff(tc.WantOutputs[i],
					have,
					cmpopts.SortSlices(func(a, b *enginev1.OutputEntry) bool { return stdcmp.Less(a.Src, b.Src) }),

```

Reply:
I've tried this (and many variations of this) in the past. I reimplemented this again now and ran the tests a few time and the same un-deterministic ordering causes test failures:

```
--- FAIL: TestCheckWithLenientScopeSearch (0.08s)
    --- FAIL: TestCheckWithLenientScopeSearch/case_21 (0.00s)
        engine_test.go:152:
                Error Trace:    /Users/samuellock/cerbos/cerbos/internal/engine/engine_test.go:152
                Error:          Should be empty, but was   (*enginev1.CheckOutput)(Inverse(protocmp.Transform, protocmp.Message{
                                        "@type":                   s"cerbos.engine.v1.CheckOutput",
                                        "actions":                 map[string]protocmp.Message{"camelCase": {"@type": s"cerbos.engine.v1.CheckOutput.ActionEffect", "effect": s"EFFECT_ALLOW", "policy": string("resource.runtime_effective_derived_roles.vdefault")}, "snake_case": {"@type": s"cerbos.engine.v1.CheckOutput.ActionEffect", "effect": s"EFFECT_ALLOW", "policy": string("resource.runtime_effective_derived_roles.vdefault")}},
                                        "effective_derived_roles": Inverse(cmpopts.SortSlices, []string{"one", "two"}),
                                        "outputs": []protocmp.Message{
                                                {
                                                        "@type": s"cerbos.engine.v1.OutputEntry",
                                                        "src": strings.Join({
                                                                "resource.runtime_effective_derived_roles.vdefault#",
                                -                               "camelC",
                                +                               "snake_c",
                                                                "ase",
                                                        }, ""),
                                                        "val": protocmp.Message{"@type": s"google.protobuf.Value", "list_value": protocmp.Message{"@type": s"google.protobuf.ListValue", "values": []protocmp.Message{{"@type": s"google.protobuf.Value", "string_value": string("one")}, {"@type": s"google.protobuf.Value", "string_value": string("two")}}}},
                                                },
                                                {
                                                        "@type": s"cerbos.engine.v1.OutputEntry",
                                                        "src": strings.Join({
                                                                "resource.runtime_effective_derived_roles.vdefault#",
                                -                               "snake_c",
                                +                               "camelC",
                                                                "ase",
                                                        }, ""),
                                                        "val": protocmp.Message{"@type": s"google.protobuf.Value", "list_value": protocmp.Message{"@type": s"google.protobuf.ListValue", "values": []protocmp.Message{{"@type": s"google.protobuf.Value", "string_value": string("one")}, {"@type": s"google.protobuf.Value", "string_value": string("two")}}}},
                                                },
                                        },
                                        "request_id":  string("test"),
                                        "resource_id": string("test"),
                                  }))
```